src/databricks/labs/lakebridge/resources/assessments/synapse/dedicated_sqlpool_extract.py

```diff
@@ -0,0 +1,141 @@
+    logger = logging.getLogger(__name__)
+
+    db_path, creds_file = arguments_loader(desc="Synapse Synapse Dedicated SQL Pool Extract Script")
+
+    cred_manager = create_credential_manager(creds_file)
+    config = cred_manager.get_credentials("synapse")["workspace"]
+
+    try:
+        synapse_workspace_settings = get_config(creds_file)["synapse"]
+        synapse_profiler_settings = synapse_workspace_settings["profiler"]
+
+        tz_info = synapse_workspace_settings["workspace"]["tz_info"]
+        workspace_tz = zoneinfo.ZoneInfo(tz_info)
+        exclude_dedicated_sql_pools = synapse_profiler_settings.get("exclude_dedicated_sql_pools", None)
+        dedicated_sql_pools_profiling_list = synapse_profiler_settings.get("dedicated_sql_pools_profiling_list", None)
+        artifacts_client = get_synapse_artifacts_client(synapse_workspace_settings)
+        workspace = SynapseWorkspace(workspace_tz, artifacts_client)
+
+        if exclude_dedicated_sql_pools:
+            msg = f"exclude_dedicated_sql_pools is set to {exclude_dedicated_sql_pools}, Skipping metrics extract for Dedicated SQL pools"
+            logger.info(msg)
+            print(json.dumps({"status": "success", "message": msg}))
+            return
+
+        dedicated_sqlpools = workspace.list_sql_pools()
+        all_dedicated_pools_list = [pool for poolPages in dedicated_sqlpools for pool in poolPages]
+        if dedicated_sql_pools_profiling_list:
+            dedicated_pools_to_profile = [
+                pool for pool in all_dedicated_pools_list if pool['name'] in dedicated_sql_pools_profiling_list
+            ]
+        else:
+            dedicated_pools_to_profile = all_dedicated_pools_list
+
+        logger.info("Pool names to extract metrics...")
+        for idx, entry in enumerate(dedicated_pools_to_profile):
+            print(f"\t {idx+1}  {entry['name'].ljust(50, '.')}{entry['status']}")
+
+        live_dedicated_pools_to_profile = [entry for entry in dedicated_pools_to_profile if entry['status'] == 'Online']
+        logger.info(f"live_dedicated_pools_to_profile: {[entry['name'] for entry in live_dedicated_pools_to_profile]}")
+
+        # Info: Extract
+        for idx, entry in enumerate(live_dedicated_pools_to_profile):
+            entry_info = f"{entry['name']} [{entry['status']}]"
+            print(f"{idx:02d})  {entry_info.ljust(60, '.')} : RUNNING extract...")
+            logger.info(f"{idx:02d})  {entry_info.ljust(60, '.')} : RUNNING extract...")
+
+            mode = "overwrite" if idx == 0 else "append"
+
+            # tables
+            table_query = SynapseQueries.list_tables()
+            connection = get_sqlpool_reader(config, entry['name'])
+            logger.info("Loading 'tables' for pool: %s", entry['name'])
+            print(f"Loading 'tables' for pool: {entry['name']}")
+            result = connection.execute(text(table_query))
+            save_resultset_to_db(result, "tables", db_path, mode=mode)
```

Comment:
This function `save_resultset_to_db()` will hold the result set open in the event of a runtime exception. As a result, when the next storage object is queried (e.g. Line 83) the connection is reused and it will always appear busy, resulting in a `Connection is busy with results for another command` error.

Reply:
I will followup with this